internal/declcfg/write.go

```diff
@@ -131,28 +135,104 @@ func writeFile(cfg DeclarativeConfig, w fsWriter, filename string) error {
 	return nil
 }
 
-func writeJSON(cfg DeclarativeConfig, w io.Writer) error {
+func WriteJSON(cfg DeclarativeConfig, w io.Writer) error {
 	enc := json.NewEncoder(w)
 	enc.SetIndent("", "    ")
 	enc.SetEscapeHTML(false)
+	return writeToEncoder(cfg, enc)
+}
+
+func WriteYAML(cfg DeclarativeConfig, w io.Writer) error {
+	enc := newYAMLEncoder(w)
+	return writeToEncoder(cfg, enc)
+}
+
+type yamlEncoder struct {
+	w io.Writer
+}
 
+func newYAMLEncoder(w io.Writer) *yamlEncoder {
+	return &yamlEncoder{w}
+}
+
+func (e *yamlEncoder) Encode(v interface{}) error {
+	var buf bytes.Buffer
+	enc := json.NewEncoder(&buf)
+	enc.SetEscapeHTML(false)
```

Comment:
Should this be allowed as a method of `yamlEncoder`? To maintain parity with the json encoder like so https://github.com/operator-framework/operator-registry/pull/641/files#diff-5e384dd70c8fb37678149443bbdc0d411af527a1839dc91a1a40cdac04fca4d5R140-R141

Reply:
That's not unreasonable. It's not _really_ necessary since we just hardcode it right now, but for the sake of symmetry, I like the suggestion.